tests/test_schema_retrieval.py

```diff
@@ -65,24 +67,63 @@ def test_invalid_schema(self):
+    def dataset_schema_url_or_path(self, request) -> str:
+        "Every kind of url or path to datasets.yaml."
+        # We use '/' instead of os.path.sep because URLs only accept / not \ as separators, but Windows path accepts
+        # both. This is not an issue for the purpose of this test.
+        return str(request.getfixturevalue('schema_file_' + request.param)) + '/datasets.yaml'
+
+    @pytest.mark.parametrize('func', (retrieve_schema_file, Schema))
+    def test_insecure_connections_fail(self, remote_dataset_schema_url, untrust_self_signed_cert, func):
         "Test insecure connections that should fail."
-        url = str(request.getfixturevalue('schema_file_' + location_type)) + '/datasets.yaml'
         with pytest.raises(InsecureConnectionError) as e:
-            retrieve_schema_file(url, tls_verification=True)
-        assert url in str(e.value)
-        assert retrieve_schema_file.__kwdefaults__['tls_verification'] is True
+            func(remote_dataset_schema_url, tls_verification=True)
+        assert remote_dataset_schema_url in str(e.value)
 
-    @pytest.mark.parametrize('location_type',
-                             ('absolute_dir',
-                              'relative_dir',
-                              'file_url',
-                              'https_url'))
-    def test_insecure_connections_succeed(self, location_type, schema_file_relative_dir, request):
-        "Test secure connections that should succeed."
+    def test_secure_connections_succeed_retrieve_schema_file(self,
+                                                             dataset_schema_url_or_path,
+                                                             schema_file_relative_dir):
+        "Test secure connections that should succeed for ``retrieve_schema_file``."
+        assert retrieve_schema_file(dataset_schema_url_or_path, tls_verification=True) == \
+            (schema_file_relative_dir / 'datasets.yaml').read_text(encoding='utf-8')
+
+    def test_secure_connections_succeed_schema(self,
+                                               dataset_schema_url_or_path,
+                                               schema_file_relative_dir):
+        "Test secure connections that should succeed for ``Schema()``."
+        assert Schema(dataset_schema_url_or_path, tls_verification=True).retrieved_url_or_path == \
+            dataset_schema_url_or_path
+
+    def test_insecure_connections_load_schemata(self, remote_dataset_schema_url, untrust_self_signed_cert):
+        "Test insecure connections that should fail when ``tls_verification=True`` for ``load_schemata``."
+        init(update_only=True, DATASET_SCHEMA_URL=remote_dataset_schema_url)
+        with pytest.raises(InsecureConnectionError) as e:
+            load_schemata(force_reload=True, tls_verification=True)
+        assert remote_dataset_schema_url in str(e.value)
+
+        # Insecure load succeeds, no exception raised
+        load_schemata(tls_verification=False)
```

Comment:
would we want to run with `force_reload=True` here too to ensure `Schema()` gets run?